### PR TITLE
Add Kafka log dump tool

### PR DESCRIPTION
**Pull request by fvaleri**
This is related to issue #5594.

I'm also attaching the output after running 1 transaction with this [test application](https://github.com/fvaleri/integ-examples/tree/main/misc/kafka/kafka-trans) on a new cluster.

We may need to add a system test, but I first wanted to get some feedback.

[my-dump.zip](https://github.com/strimzi/strimzi-kafka-operator/files/7320347/my-dump.zip)

### CODE REVIEW

**Comment by fvaleri** [comment]
@tombentley @scholzj

**Comment by fvaleri** [comment]
@tombentley what we want to do with this one?

**Comment by tombentley** [comment]
On reflection, I think it would be best to take out the dependency on `klog`, and just let people supply the partitions themselves. It makes it slightly less easy to use, but I don't really want to do a release of `klog` at this point, and having this tool depend on some binary that someone's had to build themselves doesn't feel like a great UX either.

**Comment by fvaleri** [comment]
Understood, but I don't like the idea of removing that functionality, so in my last commit I'm proposing what could be a cleaner alternative.

File: tools/log-dump/run.sh [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+    --segment       Filter by segment name
+    --out-path      Output path (default: $OUT_PATH)
+    --dry-run       Run without dumping (default: $DRY_RUN)
+    --data          Include record payload (default: $DATA)
+
+  cg_offsets      Dump consumer group offsets by group.id
+    --namespace     Kubernetes namespace
+    --cluster       Kafka cluster name
+    --group-id      Consumer group id
+    --tot-part      Consumer offsets partitions (default: $TOT_PART)
```
Comment (tombentley) [inline tools/log-dump/run.sh:249]: From this description I don't really know what `tot-part` is supposed to _be_.
```suggestion
    --tot-part      The number of __consumer_offsets partitions (default: $TOT_PART)
```

File: tools/log-dump/run.sh [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    --out-path      Output path (default: $OUT_PATH)
+  
+  txn_state       Dump transactions state by transactional.id
+    --namespace     Kubernetes namespace
+    --cluster       Kafka cluster name
+    --txn-id        Transactional id
+    --tot-part      Transaction state partitions (default: $TOT_PART)
```
Comment (tombentley) [inline tools/log-dump/run.sh:256]: Same comment, 
```suggestion
    --tot-part      The number of __transaction_state partitions (default: $TOT_PART)
```

File: tools/log-dump/run.sh [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+Usage: $0 [command] [params]
+
+  partition       Dump partition
+    --namespace     Kubernetes namespace
+    --cluster       Kafka cluster name
+    --topic         Data topic name
+    --partition     Partition number (zero-based)
+    --segment       Filter by segment name
+    --out-path      Output path (default: $OUT_PATH)
+    --dry-run       Run without dumping (default: $DRY_RUN)
```
Comment (tombentley) [inline tools/log-dump/run.sh:242]: Why is it only this subcommand that has `--dry-run`?

**Review by tombentley — APPROVED** [review]
A few last nits, but I think using `jshell` is a good compromise here.

File: tools/log-dump/run.sh (hunk 2 above)
Comment (fvaleri) [reply tools/log-dump/run.sh:256]: ok

File: tools/log-dump/run.sh (hunk 1 above)
Comment (fvaleri) [reply tools/log-dump/run.sh:249]: ok

File: tools/log-dump/run.sh (hunk 3 above)
Comment (fvaleri) [reply tools/log-dump/run.sh:242]: Good catch. In reality it is implemented for all commands. Let me fix the help.

**Review by scholzj — APPROVED** [review]

**Pull request merged**